### PR TITLE
Fixes #25762 - display filterable request UUID

### PR DESCRIPTION
**Pull request by ares**
I'd appreciate some guidance for the react code. I suppose the audit search link does not make sense to extract to a new component. What is the recommended testing approach for this code?

### CODE REVIEW

**Comment by xprazak2** [comment]
Looking good: 



I think you can leave the newly added search link where it is, unless you want to test it separately of course.

The changes are covered by the audits list snapshot already, there are relevant [test failures](https://travis-ci.org/theforeman/foreman/jobs/474138802). You will need to regenerate the snapshot once you fix the existing test errors.

**Review by lzap — DISMISSED** [review]
Nice thanks.

**Comment by ares** [comment]
I tried to understand why the js test fails, I haven't modified props or audits test, any clue what's causing it? url seems to be specified in all audit fixtures

```
● AuditsList › rendering › encountered a declaration exception
    Warning: Failed prop type: The prop `url` is marked as required in `SearchLink`, but its value is `undefined`.
        in SearchLink
      56 | const renderRequestUuidLink = (url, title, id) => {
      57 |   return (
    > 58 |     <SearchLink
         |     ^
      59 |       url={url}
      60 |       textValue={title}
      61 |       title={__('HTTP request UUID, clicking will filter audits for this request. It can also be used for searching in application logs.')}
      at CustomConsole.Object.<anonymous>.console.error (node_modules/jest-prop-type-error/index.js:5:11)
      at printWarning (node_modules/prop-types/checkPropTypes.js:19:15)
      at checkPropTypes (node_modules/prop-types/checkPropTypes.js:82:11)
      at validatePropTypes (node_modules/react/cjs/react.development.js:1686:5)
      at Object.createElementWithValidation [as createElement] (node_modules/react/cjs/react.development.js:1775:5)
      at renderRequestUuidLink (webpack/assets/javascripts/react_app/components/AuditsList/index.js:58:5)
      at renderRequestUuidLink (webpack/assets/javascripts/react_app/components/AuditsList/index.js:131:28)
      at Array.map (native)
      at map (webpack/assets/javascripts/react_app/components/AuditsList/index.js:69:13)
      at node_modules/enzyme-adapter-react-16/build/ReactSixteenAdapter.js:440:38
      at ReactShallowRenderer.render (node_modules/react-test-renderer/cjs/react-test-renderer-shallow.development.js:412:39)
      at node_modules/enzyme-adapter-react-16/build/ReactSixteenAdapter.js:444:37
      at withSetStateAllowed (node_modules/enzyme-adapter-utils/build/Utils.js:137:16)
      at Object.render (node_modules/enzyme-adapter-react-16/build/ReactSixteenAdapter.js:443:70)
      at new ShallowWrapper (node_modules/enzyme/build/ShallowWrapper.js:206:22)
      at shallow (node_modules/enzyme/build/shallow.js:21:10)
      at webpack/assets/javascripts/react_app/common/testHelpers.js:47:16
      at Array.map (native)
      at map (webpack/assets/javascripts/react_app/common/testHelpers.js:45:28)
      at Suite.<anonymous> (webpack/assets/javascripts/react_app/components/AuditsList/__tests__/AuditsList.test.js:13:24)
      at Suite.describe (webpack/assets/javascripts/react_app/components/AuditsList/__tests__/AuditsList.test.js:12:3)
      at Object.describe (webpack/assets/javascripts/react_app/components/AuditsList/__tests__/AuditsList.test.js:11:1)
```

**Comment by ares** [comment]
Thanks to @xprazak2 I was able to figure out that fixtures need updating, tests should be green now

As part of this, I realized `SearchLink` requires passed id, which is the number. It would make more sense to pass the whole string for html id, this way it's easy to generate 2 elements with the same id, as it does `resource-link-{id}`. It's out of the scope for this PR though.

**Comment by ares** [comment]
Now, eslint errors, will check tomorrow

**Comment by ares** [comment]
eslint should be now fixed

**Review by xprazak2 — DISMISSED** [review]
Works, js tests are green now and the jenkins failure is unrelated. Since we are in stabilization period, I leave it to @tbrisker to click the button or wait until we branch.

File: webpack/assets/javascripts/react_app/components/AuditsList/index.js [hunk 1]
```diff
@@ -52,7 +52,18 @@ const renderResourceLink = (auditTitle, auditTitleUrl, id) => {
...
+    url={url}
+    textValue={title}
+    title={__(
+      'HTTP request UUID, clicking will filter audits for this request. It can also be used for searching in application logs.'
+    )}
+    id={id}
+  />
+);
+
+const AuditsList = ({ data: { audits, isOrgEnabled, isLocEnabled } }) => (
```
Comment (ohadlevy) [inline webpack/assets/javascripts/react_app/components/AuditsList/index.js]: arn't org / loc will be always enabled in 1.21?

File: webpack/assets/javascripts/react_app/components/AuditsList/index.js [hunk 2]
```diff
@@ -107,6 +120,21 @@ const AuditsList = ({ data: { audits } }) => (
             <ActionLinks allowedActions={allowedActions} />
           </Row>
 
+          <Row>
+            <Col sm={10} className="request_uuid_column">
+              <Row>
+                <Col md={2}>
+                  <span>Request UUID</span>
```
Comment (ohadlevy) [inline webpack/assets/javascripts/react_app/components/AuditsList/index.js]: i18n?

**Review by ohadlevy — CHANGES_REQUESTED** [review]
minor nit - thanks

File: webpack/assets/javascripts/react_app/components/AuditsList/index.js (hunk 1 above)
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/AuditsList/index.js]: Looks like bad rebase, the `isOrgEnabled` and `isLocEnabled` parameters have been removed in c949b20a5d64a906eb6d9289755de33463c4e199

**Comment by ares** [comment]
Ok, thanks for comments, dropped orgs/locs switch, that was bad rebase and added string extraction. Please take another look.

File: webpack/assets/javascripts/react_app/components/AuditsList/audit.scss [hunk 3]
```diff
@@ -105,7 +105,8 @@
       }
     }
 
-    .taxonomy-section {
+    .taxonomy-section,
+    .request_uuid_column {
```
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/AuditsList/audit.scss:109]: nitpick: the standard for class naming is `.request-uuid-column`.

File: webpack/assets/javascripts/react_app/components/AuditsList/index.js [hunk 4]
```diff
@@ -107,6 +120,21 @@ const AuditsList = ({ data: { audits } }) => (
             <ActionLinks allowedActions={allowedActions} />
           </Row>
 
+          <Row>
```
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/AuditsList/index.js:123]: I think this file is getting big.
Would you mind to separate it to a new component `AuditList/ItemRequestUuidRow`?
In this case, there is no reason to use `renderRequestUuidLink` and you can just embed it inside your new component.

File: webpack/assets/javascripts/react_app/components/AuditsList/index.js [hunk 5]
```diff
@@ -107,6 +120,21 @@ const AuditsList = ({ data: { audits } }) => (
...
+            <Col sm={10} className="request_uuid_column">
+              <Row>
+                <Col md={2}>
+                  <span>{__('Request UUID')}</span>
+                </Col>
+                <Col md={10}>
+                  <strong>
```
Comment (sharvit) [inline webpack/assets/javascripts/react_app/components/AuditsList/index.js:130]: Could you use css to make it strong?
It's not a good practice to render big html dom trees inside `strong` or `b`.

```css
font-weight: bold;
```

**Review by sharvit — CHANGES_REQUESTED** [review]

**Comment by ohadlevy** [comment]
@ares some comments are unanswered + rebase is now required.

**Comment by ezr-ondrej** [comment]
@ares what is your plan here? I could try to take over, if you don't have time to finish it.

**Comment by ares** [comment]
@ezr-ondrej that would be great, I struggle to finish it, though majority of it should be done

**Comment by ares** [comment]
@ezr-ondrej is there a chance you take this over? otherwise, I'll close as I'm unable to finish it.

**Comment by ezr-ondrej** [comment]
> @ezr-ondrej is there a chance you take this over? otherwise, I'll close as I'm unable to finish it.

Closing in favor of #7369 thanks @ares !
Sorry it took a while :upside_down_face:

**Pull request closed without merging**